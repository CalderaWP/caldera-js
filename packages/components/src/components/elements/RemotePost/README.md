# Name Of Remote Post
Displays a WordPress post.

Notes:
* This component expects a WordPress post as returned by the WordPress REST API. It does not fetch the post.
* This component uses [humanmade/react-oembed-container](https://github.com/humanmade/react-oembed-container) for handling oembeds, HTML, etc.

## Usage

### Import With webpack
`import {RemotePost} from '@calderawp/components';`

### Show Full Content Of Post

The prop `showFullContent`, if set to true, causes the component to display the full post content `post.content.rendered`
```jsx
import {RemotePost} from '@calderawp/components';
<RemotePost 
    post={{}}
    showFullContent={true}
/>

```

### Show Excerpt Only
The prop `showFullContent`, if set to false, which is its default value, causes the component to display the post excerpt - `post.excerpt.rendered`

```jsx
<RemotePost 
    post={{}}
/>

