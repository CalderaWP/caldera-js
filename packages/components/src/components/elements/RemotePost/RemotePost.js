import React, { Fragment } from 'react';
import { postPropTypes } from './propTypes';
import EmbedContainer from 'react-oembed-container';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * Show a WordPress post from the REST API with embeds allowed
 *
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export const RemotePost = ({
	post, //WordPress post
	showFullContent, //false show excerpt
	className //extra class name for wrapper component
}) => {
	return (
		<EmbedContainer markup={post.content.rendered}>
			<article
				className={classNames(
					className,
					post.type,
					`post-${post.id}`,
					`type-${post.type}`,
					`status-${post.status}`,
					'hentry entry' //I need to return post_class or something from REST API??
				)}
				id={`post-${post.id}`}
			>
				<header className="entry-header">
					<h1 className="entry-title">{post.title.rendered}</h1>
				</header>
				<Fragment>
					{showFullContent ? (
						<div
							className="entry-content"
							dangerouslySetInnerHTML={{
								__html: post.content.rendered
							}}
						/>
					) : (
						<div
							className="entry-excerpt"
							dangerouslySetInnerHTML={{
								__html: post.excerpt.rendered
							}}
						/>
					)}
				</Fragment>
			</article>
		</EmbedContainer>
	);
};

RemotePost.propTypes = {
	...postPropTypes,
	className: PropTypes.string,
	showFullContent: PropTypes.bool
};

RemotePost.defaultPosts = {
	showFullContent: false
};
