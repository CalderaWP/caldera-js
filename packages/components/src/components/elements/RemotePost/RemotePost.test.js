import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { RemotePost } from './RemotePost';
Enzyme.configure({ adapter: new Adapter() });

const post = {
	id: 42,
	title: {
		rendered: 'Hello To Roy'
	},
	content: {
		rendered: '<p>Hi Roy</p>'
	},
	excerpt: {
		rendered: '<p>Hi From Excerpt</p>'
	}
};
describe('Remote post  Component', () => {
	it('Matches snapshot', () => {
		const component = renderer.create(<RemotePost post={post} />);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Matches snapshot with class names', () => {
		const component = renderer.create(
			<RemotePost post={post} className={'remote-post-for-help'} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Matches snapshot with full content', () => {
		const component = renderer.create(
			<RemotePost
				post={post}
				showFullContent={true}
				className={'remote-post-for-help'}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
