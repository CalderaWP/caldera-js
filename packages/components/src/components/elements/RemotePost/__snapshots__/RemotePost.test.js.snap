// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remote post  Component Matches snapshot 1`] = `
<div
  className={null}
>
  <article
    className="post-42 type-undefined status-undefined hentry entry"
    id="post-42"
  >
    <header
      className="entry-header"
    >
      <h1
        className="entry-title"
      >
        Hello To Roy
      </h1>
    </header>
    <div
      className="entry-excerpt"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hi From Excerpt</p>",
        }
      }
    />
  </article>
</div>
`;

exports[`Remote post  Component Matches snapshot with class names 1`] = `
<div
  className={null}
>
  <article
    className="remote-post-for-help post-42 type-undefined status-undefined hentry entry"
    id="post-42"
  >
    <header
      className="entry-header"
    >
      <h1
        className="entry-title"
      >
        Hello To Roy
      </h1>
    </header>
    <div
      className="entry-excerpt"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hi From Excerpt</p>",
        }
      }
    />
  </article>
</div>
`;

exports[`Remote post  Component Matches snapshot with full content 1`] = `
<div
  className={null}
>
  <article
    className="remote-post-for-help post-42 type-undefined status-undefined hentry entry"
    id="post-42"
  >
    <header
      className="entry-header"
    >
      <h1
        className="entry-title"
      >
        Hello To Roy
      </h1>
    </header>
    <div
      className="entry-content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hi Roy</p>",
        }
      }
    />
  </article>
</div>
`;
